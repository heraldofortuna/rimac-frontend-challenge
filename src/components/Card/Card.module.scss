.card {
  background-color: $neutrals-0-color;
  padding: 40px 24px;
  border-radius: 24px;
  box-shadow: 0 1px 32px 0 rgba(174, 172, 243, 0.35);
}