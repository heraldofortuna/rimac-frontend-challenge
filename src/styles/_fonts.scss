@font-face {
  font-family: 'BR Sonoma Regular';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular'), url('/fonts/BRSonoma-Regular-BF654c45266c042.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Regular Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular Italic'), url('/fonts/BRSonoma-RegularItalic-BF654c452681c11.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin'), url('/fonts/BRSonoma-Thin-BF654c45255ffe0.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin Italic'), url('/fonts/BRSonoma-ThinItalic-BF654c45268d3f9.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight'), url('/fonts/BRSonoma-ExtraLight-BF654c45265af8d.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight Italic'), url('/fonts/BRSonoma-ExtraLightItalic-BF654c4525a5046.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light'), url('/fonts/BRSonoma-Light-BF654c452608e0f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light Italic'), url('/fonts/BRSonoma-LightItalic-BF654c45266aa83.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium'), url('/fonts/BRSonoma-Medium-BF654c45266edd1.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium Italic'), url('/fonts/BRSonoma-MediumItalic-BF654c45267d45f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold'), url('/fonts/BRSonoma-SemiBold-BF654c45268c340.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold Italic'), url('/fonts/BRSonoma-SemiBoldItalic-BF654c452696350.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold'), url('/fonts/BRSonoma-Bold-BF654c4526823f5.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold Italic'), url('/fonts/BRSonoma-BoldItalic-BF654c4525c9c27.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black'), url('/fonts/BRSonoma-Black-BF654c4525506bf.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black Italic'), url('/fonts/BRSonoma-BlackItalic-BF654c45268988e.woff') format('woff');
}