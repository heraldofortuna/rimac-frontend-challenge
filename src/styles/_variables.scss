$grey-100-color: #03050F;
$grey-80-color: #2B304E;
$grey-30-color: #CCD1EE;
$grey-10-color: #F8F9FF;
$blue-berry-600-color: #4F4FFF;
$neutrals-700-color: #141938;
$neutrals-600-color: #7981B2;
$neutrals-0-color: #FFFFFF;

$sonoma-font: 'BR Sonoma Regular', sans-serif;
$lato-font: 'Lato', sans-serif;

$breakpoint-mobile: 480px;
$breakpoint-desktop: 768px;