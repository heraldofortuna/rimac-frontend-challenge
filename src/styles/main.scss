@use 'fonts';
@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700;900&display=swap');

* {
  padding: 0;
  margin: 0;
}

body {
  background-color: $grey-10-color;
  font-family: $sonoma-font;
  color: $grey-100-color;
}

#root {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}