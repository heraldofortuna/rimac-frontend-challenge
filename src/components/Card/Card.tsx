import type { ICardProps } from "../../types/components/card";
import CheckIcon from "@assets/check.svg";
import styles from "./Card.module.scss";

const Card: React.FC<ICardProps> = ({
  children,
  isSelectable = false,
  isSelected = false,
  onClick = () => {},
}) => {
  const cardClasses = `${styles.card} ${isSelectable ? styles.selectable : ""} ${isSelected ? styles.selected : ""}`;
  const handleClick = isSelectable ? onClick : undefined;

  return (
    <div className={cardClasses} onClick={handleClick}>
      {isSelectable ? (
        isSelected ? (
          <img
            src={CheckIcon}
            alt="Card seleccionado"
            className={styles.card__check}
          />
        ) : (
          <span className={styles.card__circle} data-testid="selection-circle"></span>
        )
      ) : (
        <></>
      )}
      {children}
    </div>
  );
};

export default Card;
