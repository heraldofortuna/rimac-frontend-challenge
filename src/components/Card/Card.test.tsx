import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import Card from "./Card";
import type { ICardProps } from "../../types/components/card";
import styles from "./Card.module.scss";

describe("Card component", () => {
  const defaultProps: ICardProps = {
    children: <div>Card Content</div>,
    isSelectable: false,
    isSelected: false,
    onClick: vi.fn(),
  };

  const renderCard = (props: Partial<ICardProps> = {}) => {
    return render(<Card {...defaultProps} {...props} />);
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders correctly with default props", () => {
    renderCard();
    
    const card = screen.getByText("Card Content").parentElement;
    expect(card).toBeInTheDocument();
    expect(card).toHaveClass(styles.card);
    expect(card).not.toHaveClass(styles.selectable);
    expect(card).not.toHaveClass(styles.selected);
    
    expect(screen.queryByAltText("Card seleccionado")).not.toBeInTheDocument();
    expect(screen.queryByTestId("selection-circle")).not.toBeInTheDocument();
  });

  it("renders selectable card correctly", () => {
    renderCard({ isSelectable: true });
    
    const card = screen.getByText("Card Content").parentElement;
    expect(card).toHaveClass(styles.card);
    expect(card).toHaveClass(styles.selectable);
    expect(card).not.toHaveClass(styles.selected);
    
    const circle = screen.getByTestId("selection-circle");
    expect(circle).toBeInTheDocument();
    expect(circle).toHaveClass(styles.card__circle);
  });

  it("renders selected card correctly", () => {
    renderCard({ isSelectable: true, isSelected: true });
    
    const card = screen.getByText("Card Content").parentElement;
    expect(card).toHaveClass(styles.card);
    expect(card).toHaveClass(styles.selectable);
    expect(card).toHaveClass(styles.selected);
    
    const checkIcon = screen.getByAltText("Card seleccionado");
    expect(checkIcon).toBeInTheDocument();
    expect(checkIcon).toHaveClass(styles.card__check);
  });

  it("calls onClick handler when clicked", async () => {
    const handleClick = vi.fn();
    renderCard({ isSelectable: true, onClick: handleClick });
    
    const card = screen.getByText("Card Content").parentElement;
    await fireEvent.click(card!);
    
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("does not call onClick when not selectable", async () => {
    const handleClick = vi.fn();
    renderCard({ isSelectable: false, onClick: handleClick });
    
    const card = screen.getByText("Card Content").parentElement;
    fireEvent.click(card!);

    expect(handleClick).not.toHaveBeenCalled();
  });

  it("applies correct classes based on props", () => {
    const { rerender } = renderCard();
    let card = screen.getByText("Card Content").parentElement;
    
    expect(card).toHaveClass(styles.card);
    expect(card).not.toHaveClass(styles.selectable);
    expect(card).not.toHaveClass(styles.selected);
    
    rerender(<Card {...defaultProps} isSelectable={true} />);
    card = screen.getByText("Card Content").parentElement;
    expect(card).toHaveClass(styles.card);
    expect(card).toHaveClass(styles.selectable);
    expect(card).not.toHaveClass(styles.selected);

    rerender(<Card {...defaultProps} isSelectable={true} isSelected={true} />);
    card = screen.getByText("Card Content").parentElement;
    expect(card).toHaveClass(styles.card);
    expect(card).toHaveClass(styles.selectable);
    expect(card).toHaveClass(styles.selected);
  });

  it("matches snapshot for non-selectable card", () => {
    const { asFragment } = renderCard();
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for selectable card", () => {
    const { asFragment } = renderCard({ isSelectable: true });
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for selected card", () => {
    const { asFragment } = renderCard({ isSelectable: true, isSelected: true });
    expect(asFragment()).toMatchSnapshot();
  });
});