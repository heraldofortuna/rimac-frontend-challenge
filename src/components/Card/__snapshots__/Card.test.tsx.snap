// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > matches snapshot for non-selectable card 1`] = `
<DocumentFragment>
  <div
    class="_card_526f9d  "
  >
    <div>
      Card Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card component > matches snapshot for selectable card 1`] = `
<DocumentFragment>
  <div
    class="_card_526f9d _selectable_526f9d "
  >
    <span
      class="_card__circle_526f9d"
      data-testid="selection-circle"
    />
    <div>
      Card Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card component > matches snapshot for selected card 1`] = `
<DocumentFragment>
  <div
    class="_card_526f9d _selectable_526f9d _selected_526f9d"
  >
    <img
      alt="Card seleccionado"
      class="_card__check_526f9d"
      src="data:image/svg+xml,%3csvg%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M12%200C5.4%200%200%205.4%200%2012C0%2018.6%205.4%2024%2012%2024C18.6%2024%2024%2018.6%2024%2012C24%205.4%2018.6%200%2012%200Z'%20fill='%23389E0D'/%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M17.1813%207.11293C17.5607%207.434%2017.6081%208.00188%2017.287%208.38133L10.687%2016.1813C10.5215%2016.3769%2010.2806%2016.4927%2010.0245%2016.4996C9.76844%2016.5066%209.52153%2016.4042%209.34564%2016.2179L5.94564%2012.6179C5.60435%2012.2566%205.62063%2011.687%205.98199%2011.3457C6.34336%2011.0044%206.91297%2011.0207%207.25427%2011.382L9.96311%2014.2502L15.9129%207.21863C16.234%206.83919%2016.8019%206.79187%2017.1813%207.11293Z'%20fill='white'/%3e%3c/svg%3e"
    />
    <div>
      Card Content
    </div>
  </div>
</DocumentFragment>
`;
